src/stores/callAPIMiddleware.js

```diff
@@ -19,30 +19,29 @@ export function callAPIMiddleware({ dispatch, getState }) {
       type: types.request,
     });
 
-    return doPerform(getState(), dispatch)
-      .then((response) => {
-        dispatch({
-          response,
-          type: types.success,
-        });
-
-        return response;
-      })
-      .catch((error) => {
-        dispatch({
-          error: error.message,
-          type: types.error,
-        });
-
-        if (throwOnError) {
-          throw error;
-        }
-
-        // TODO: Remove this, this supports a pattern we want to factor out
-        // eventually.
-        return {
-          error: error.message,
-        };
+    try {
```

Comment:
The problem here was when the `shouldPerform()` check returned `false`. The result would've been `undefined`, which has no `then()` method, and then resulted in an error.